tools/chpl-language-server/src/chpl-language-server.py

```diff
@@ -1020,6 +1066,31 @@ def _setup_regexes(self):
         # use pat2 first since it is more specific
         self._find_rename_deprecation_regex = re.compile(f"({pat2})|({pat1})")
 
+    def _setup_linter(self, clsConfig: CLSConfig):
+        """
+        Setup the linter, if it is enabled
+        """
+        if not (
+            self.do_linting
+            and chplcheck
+            and chplcheck_lsp
+            and chplcheck_config
+            and chplcheck_driver
+            and chplcheck_rules
+        ):
```

Comment:
Using a single object that is required to have all of the modules would help simplify this conditional. If you take my suggestion of creating only a single global object, then the "all of the modules" condition will automatically be satisfied because the `chplcheck: ChplcheckProxy` variable will only be initialized if no exceptions were thrown at all, including when loading each of the modules.